snuba/datasets/errors.py

```diff
@@ -0,0 +1,302 @@
+from datetime import timedelta
+from typing import Mapping, Sequence, Union
+
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnSet,
+    DateTime,
+    FixedString,
+    IPv4,
+    IPv6,
+    LowCardinality,
+    Materialized,
+    Nested,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+    WithCodecs,
+    WithDefault,
+)
+from snuba.datasets.dataset import ColumnSplitSpec, TimeSeriesDataset
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.table_storage import TableWriter, KafkaStreamLoader
+from snuba.datasets.errors_processor import ErrorsProcessor
+from snuba.datasets.schemas.tables import ReplacingMergeTreeSchema
+from snuba.datasets.tags_column_processor import TagColumnProcessor
+from snuba.query.processors.basic_functions import BasicFunctionsProcessor
+from snuba.query.processors.prewhere import PrewhereProcessor
+from snuba.query.query import Query
+from snuba.query.extensions import QueryExtension
+from snuba.query.parsing import ParsingContext
+from snuba.query.query_processor import QueryProcessor
+from snuba.query.timeseries import TimeSeriesExtension
+from snuba.query.project_extension import ProjectExtension, ProjectWithGroupsProcessor
+
+
+class ErrorsDataset(TimeSeriesDataset):
+    """
+    Represents the collections of all event types that are not transactions.
+
+    This is meant to replace Events. They will both exist during the migration.
+    """
+
+    def __init__(self) -> None:
+        metadata_columns = ColumnSet(
+            [
+                ("offset", WithCodecs(UInt(64), ["DoubleDelta", "LZ4"])),
+                ("partition", UInt(16)),
+            ]
+        )
+
+        promoted_tag_columns = ColumnSet(
+            [
+                # These are the classic tags, they are saved in Snuba exactly as they
+                # appear in the event body.
+                ("level", LowCardinality(String())),
+                # ("logger", Nullable(String())),
+                # ("server_name", Nullable(String())),  # future name: device_id?
+                ("environment", LowCardinality(Nullable(String()))),
+                # ("sentry:release", Nullable(String())),
+                ("release", LowCardinality(Nullable(String()))),
+                # ("sentry:dist", Nullable(String())),
+                ("dist", LowCardinality(Nullable(String()))),
+                # ("sentry:user", Nullable(String())),
+                ("user", WithDefault(String(), "''")),
+                # ("site", Nullable(String())),
+                # ("url", Nullable(String())),
+                ("transaction_name", LowCardinality(String())),
+            ]
+        )
+
+        required_columns = ColumnSet(
+            [
+                ("event_id", WithCodecs(UUID(), ["NONE"])),
```

Comment:
Do we want organization_id in the new schema? It might help with cross project queries.

Reply:
Not yet. project can still be reassigned to different orgs.

Reply:
Booooooo